src/controllers/serverRoomIDController.ts

```diff
@@ -111,24 +111,48 @@ export const createRoom = async (req: Request, res: Response) => {
 // Get all rooms
 export const getAllRooms = async (req: Request, res: Response) => {
   try {
-    const rooms = await RoomId.find({});
+    const { query } = req.query;
+
+    let roomsQuery = {};
+
+    if (query) {
+      roomsQuery = {
+        $or: [
+          { roomId: { $regex: query, $options: 'i' } },
```

Comment:
search will work not work on id, password these are secure parameters

Reply:
ok